src/main/java/org/skriptlang/skript/bukkit/recipes/elements/ExprAllRecipes.java

```diff
@@ -0,0 +1,131 @@
+package org.skriptlang.skript.bukkit.recipes.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.RecipeUtils;
+import ch.njol.util.Kleenean;
+import org.bukkit.Bukkit;
+import org.bukkit.Keyed;
+import org.bukkit.NamespacedKey;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.Recipe;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+@Name("All Recipes")
+@Description({
+	"Retrieve all recipes registered in the server",
+	"<ul>",
+	"<li>You can retrieve all recipes of a recipe type.</li>",
+	"<li>You can retrieve all recipes of an item, including custom items.</li>",
+	"<li>You can retrieve all minecraft recipes.</li>",
+	"<li>You can retrieve all custom recipes made by any and all plugins.</li>",
+	"</ul>"
+})
+@Examples({
+	"set {_list::*} to all of the recipe of type shaped recipe for netherite ingot",
+	"set {_list::*} to all mc recipes of type cooking recipe for raw beef",
+	"set {_list::*} to all of the custom recipes of type blasting for raw iron named \"Impure Iron\""
+})
+@Since("INSERT VERSION")
+public class ExprAllRecipes extends SimpleExpression<Recipe> {
+
+	static {
+		Skript.registerExpression(ExprAllRecipes.class, Recipe.class, ExpressionType.SIMPLE,
+			"all [of the] recipes [items:for %-itemstacks/itemtypes%]",
+			"all [of the] (mc|minecraft|vanilla) recipes [items:for %-itemstacks/itemtypes%]",
+			"all [of the] custom recipes [items:for %-itemstacks/itemtypes%]",
+			"all [of the] %recipetype% [items:for %-itemstacks/itemtypes%]",
+			"all [of the] (mc|minecraft|vanilla) %recipetype% [items:for %-itemstacks/itemtypes%]",
+			"all [of the] custom %recipetype% [items:for %-itemstacks/itemtypes%]");
+	}
+
+	private Expression<RecipeUtils.RecipeType> recipeTypeExpr;
+	private Expression<?> itemExpr;
+	private boolean getMinecraft, getCustom;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		//noinspection unchecked
+		recipeTypeExpr = matchedPattern >= 3 ? (Expression<RecipeUtils.RecipeType>) exprs[0] : null;
+		itemExpr = parseResult.hasTag("items") ? exprs[1] : null;
```

Comment:
the parse tag isn't necessary, and also this will error with array out of bounds on the first and third patterns! you should be testing this as you go!